podcasts/PodcastImageView.swift

```diff
@@ -28,8 +30,21 @@ class PodcastImageView: UIView {
     func setEpisode(_ episode: Episode, size: PodcastThumbnailSize) {
         guard let imageView = imageView else { return }
 
+        currentEpisode = episode
+
         Task {
             if FeatureFlag.episodeFeedArtwork.enabled, Settings.loadEmbeddedImages, let episodeArtworkUrl = try? await ShowInfoCoordinator.shared.loadEpisodeArtworkUrl(podcastUuid: episode.parentIdentifier(), episodeUuid: episode.uuid) {
+
+                ImageManager.sharedManager.setPlaceholder(imageView: imageView, size: size)
+                imageView.kf.cancelDownloadTask()
+
+                // The app might run into the case where the episode changed but there's still
+                // a pending task to display the image of another episode
+                // This check prevent this from happening.
```

Comment:
I wonder if this comment is conveying the message in a good way, feedback is appreciated.

Basically, this prevents something like:

1. We start to load episode artwork for episode A
2. The user scroll downs and the cell get changed to episode B
3. We already have episode B metadata and image cached
4. Episode B image is shown
5. Episode artwork for episode A finishes loading, so the executing proceeds to load its image
6. 💥 Episode artwork for episode A is shown while what's being shown is episode B